awx/main/conf.py

```diff
@@ -765,6 +765,17 @@
     help_text=_('Indicates whether the instance is part of a kubernetes-based deployment.'),
 )
 
+register(
+    'SUBSCRIPTION_COMPLIANCE_WITH_DELETING',
+    field_class=fields.BooleanField,
+    default=False,
+    label=_('Involve deleted hosts into Compliance computation and shows Host Metric menu.'),
+    help_text=_('If set to true, count of Hosts Automated is decreased by number of deleted host metrics.'),
+    read_only=True,
```

Comment:
@TheRealHaoLiu is the `read_only` flag ok?
If I set it here, the value in `awx/settings/defaults.py` and `awx/settings/local_overrides.py` is not relevant

Reply:
Settings defined here have very non-obvious precedence rules. If the user sets it in a settings file then it will automatically become read-only. IIRC, we very infrequently use the read_only setting here and I suspect it will have some bugs.

Reply:
@slemrmartin @TheRealHaoLiu I would maybe do this in a more future proof way, I'd say setting:

SUBSCRIPTION_USAGE_MODEL (string)

The value we'd be setting now would be "unique_managed_hosts", which affects the bacground processing and if we'll show/compute the compliance status and used/remaining hosts. (so only AoC could set this to None or to "unique_managed_hosts_montly" directly)

-----

In 2.5 we'd do  "unique_managed_hosts_montly", that will only compute monthly usage, but will  hide remaining and compliance (since it's usage based, there is no upper limit)

------

and lets send this in Analytics under the cluster_info.json

Reply:
added to `config.json` next to the `free_instances` value

Reply:
https://gitlab.cee.redhat.com/automation-analytics/automation-analytics-backend/-/merge_requests/697